src/app/zone-ingresses/data/index.spec.ts

```diff
@@ -0,0 +1,287 @@
+import mock from '@/test-support/mocks/src/zone-ingresses/_/_overview'
+import { ZoneIngressOverview as PartialZoneIngressOverview } from '@/types/index'
+
+type Writeable<T> = { -readonly [P in keyof T]: Writeable<T[P]> };
+
+type Assert = (actual: ZoneIngressOverview, item: PartialZoneIngressOverview) => void
+type Context = {
+  message: string
+  assert: Assert
+  get: () => Promise<PartialZoneIngressOverview>
+}
+type Test = {
+  env?: Record<string, string>
+  message: string
+  assert: Assert
+  setup: (item: PartialZoneIngressOverview) => Promise<PartialZoneIngressOverview>
+}
+
+const freeze = (obj: any) => {
+  Object.keys(obj).forEach(prop => {
+    if (typeof obj[prop] === 'object' && !Object.isFrozen(obj[prop])) {
+      freeze(obj[prop])
+    }
+  })
+  return Object.freeze(obj)
+}
+
+const get = async (env: Record<string, string>) => {
+  const responder = useResponder({
+    _: mock,
+  }, (key: string, d = '') => env[key] ?? d)
+  const request = responder('_')
+  return (await request(
+    {
+      method: 'GET',
+      body: {},
+      url: {
+        searchParams: new URLSearchParams(),
+      },
+      params: {
+        name: 'zone',
+      },
+    },
+  )).body
+}
+
+const map = (test: Test) => {
+  return {
+    message: test.message,
+    assert: test.assert,
+    get: async () => {
+      const item = (await get(test.env ?? {})) as unknown as Writeable<PartialZoneIngressOverview>
+      return freeze(await test.setup(item)) as Readonly<PartialZoneIngressOverview>
+    },
+  }
+}
+const run = async ({ get, assert }: Context) => {
+  const item = await get()
+  assert(ZoneIngressOverview.fromObject(item), item)
+}
+//
+describe('ZoneIngressOverview', () => {
+  describe('zoneInsight.subscriptions', () => {
+    test.each(([
+      {
+        message: 'absent zoneInsight remains undefined',
+        setup: async (item) => {
+          delete item.zoneIngressInsight
+          return item
+        },
+        assert: (actual) => {
+          expect(actual.zoneIngressInsight).toBeUndefined()
+        },
+      },
+      {
+        message: 'no subscriptions, connectedSubscription remains undefined',
+        setup: async (item) => {
+          if (typeof item.zoneIngressInsight !== 'undefined') {
+            item.zoneIngressInsight.subscriptions = []
+          }
+          return item
+        },
+        assert: (actual) => {
+          expect(actual.zoneIngressInsight).toBeDefined()
+          expect(actual.zoneIngressInsight?.subscriptions.length).toStrictEqual(0)
+          expect(actual.zoneIngressInsight?.connectedSubscription).toBeUndefined()
+        },
+      },
+      {
+        message: 'all disconnected subscriptions, connectedSubscription remains undefined',
+        env: {
+          KUMA_SUBSCRIPTION_COUNT: '10',
+        },
+        setup: async (item) => {
+          if (typeof item.zoneIngressInsight !== 'undefined') {
+            item.zoneIngressInsight.subscriptions.forEach((item: any) => {
```

Comment:
**Question**: Didn’t you say you were handling https://github.com/kumahq/kuma-gui/issues/1674 as part of the data layer work on Zones?

Reply:
Yes, but we can do that once all of this work is in. This PR is about testing so I'm working with what we have right now.

Reply:
Same again here as a above, just to note in case, this file is for comparison only